ee/audit/run_ee.go

```diff
@@ -107,37 +107,161 @@ func run() error {
 		glog.Info("audit file is empty")
 		return nil
 	}
-	var iterator int64 = 0
 
-	iv := make([]byte, aes.BlockSize)
-	x.Check2(file.ReadAt(iv, iterator))
-	iterator = iterator + aes.BlockSize
+	// decrypt header in audit log to verify encryption key
+	// [16]byte IV + [4]byte len(x.VerificationText) + [11]byte x.VerificationText
+	decryptHeader := func() ([]byte, int64, error) {
```

Comment:
What is the purpose of making them the same function?  For the updated audit logs the header looks like
```
[16]byte IV + [4]byte len(x.VerificationText) + [11]byte x.VerificationText
```

whereas for the old audit logs the header is
```
[12]byte baseIV + [4]byte len(x.VerificationTextDeprecated) + [11]byte x.VerificationTextDeprecated
```

and therefore they need to be parsed differently.  It is possible that we would have to try both functions to decrypt the header, so it would be very confusing to try to put the parsing logic of both inside one function.

Reply:
Additionally the functions `decryptBody` and `decryptBodyDeprecated` have very different parsing logic (the way they increment and get the length of text is fundamentally different).  They should not be written in one function.

Reply:
As discussed we will refactor to use io.Reader object

Reply:
Refactored to use `file io.ReaderAt, outfile io.Writer`